srml/slashing/src/tests.rs

```diff
@@ -0,0 +1,161 @@
+#![cfg(test)]
+
+use super::*;
+use runtime_io::with_externalities;
+use srml_support::{decl_module, impl_outer_origin, dispatch::Result};
+use system::{self, ensure_signed};
+use substrate_primitives::{Blake2Hasher, H256};
+use primitives::{
+	BuildStorage,
+	traits::{BlakeTwo256, IdentityLookup},
+	testing::{Digest, DigestItem, Header}
+};
+
+impl_outer_origin! {
+	pub enum Origin for Test {}
+}
+
+type Balance = balances::Module<Test>;
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct Test;
+
+impl system::Trait for Test {
+	type Origin = Origin;
+	type Index = u32;
+	type BlockNumber = u64;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type Digest = Digest;
+	type AccountId = u64;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = ();
+	type Log = DigestItem;
+}
+
+impl balances::Trait for Test {
+	type Balance = u64;
+	type OnFreeBalanceZero = ();
+	type OnNewAccount = ();
+	type Event = ();
+	type TransactionPayment = ();
+	type TransferPayment = ();
+	type DustRemoval = ();
+}
+
+impl OnSlashing for Test {
+	type Currency = Balance;
+	type Misconduct = Unresponsive;
+	type Slashing = Self;
+
+	fn slash(
+		who: &Self::AccountId,
+		free_balance: BalanceOf<Self>,
+		misconduct: &mut Self::Misconduct
+	) -> (NegativeImbalanceOf<Self>, BalanceOf<Self>) {
+		let slashed_amount = Self::Slashing::amount(free_balance, misconduct.severity());
+		let (imbalance, remaining) = Self::Currency::slash(&who, slashed_amount);
+		misconduct.on_misconduct();
+		(imbalance, remaining)
+	}
+
+	fn on_signal(misconduct: &mut Self::Misconduct) {
+		misconduct.on_signal();
+	}
+}
+
+impl Slashing for Test {
```

Comment:
This might be simply because of lack of docs (which is expected given it is a draft): 
- My expectation was between `Slashing` and `OnSlashing` one wraps the other so both won't have to be implemented? Maybe they could have a clearer distinction on the documentation in the future.
- And the function itself? I can't find it anywhere in trait definitions.